qcodes/instrument/parameter.py

```diff
@@ -533,28 +533,34 @@ def _from_raw_value_to_value(self, raw_value: ParamRawDataType
         # apply offset first (native scale)
         if self.offset is not None:
             # offset values
-            if isinstance(self.offset, collections.abc.Iterable):
-                # offset contains multiple elements, one for each value
-                value = tuple(val - offset for val, offset
-                              in zip(value, self.offset))
-            elif isinstance(value, collections.abc.Iterable):
-                # Use single offset for all values
-                value = tuple(val - self.offset for val in value)
-            else:
-                value -= self.offset
+            try:
+                value = value - self.offset
```

Comment:
@Akshita07 my main comment is: i think we should also mirror a part of this change into the ``_from_value_to_raw_value`` method, so that setting a numpy-valued parameter with not-None offset/scale works as well. the only change needed for that is to change ``raw_value *= self.scale`` to ``raw_value = raw_value * self.scale`` and ``raw_value += self.offset`` to ``raw_value = raw_value + self.offset``. would you agree @jenshnielsen ?

Reply:
Yes, I'll update it.